base_decorator.rb

```diff
@@ -0,0 +1,25 @@
+require './namable_class'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    name = @nameable.correct_name.capitalize
+    name.length > 10 ? name[0..9] : name
+  end
+end
```

Comment:
- [ ] In reading your project requirements, it says: "For the TrimmerDecorator:
Create a class that inherits from the base Decorator class."
Therefore, I would advise you to create TrimmerDecorator class separately and move this code inside.